packages/firebase_ml_vision/lib/src/text_detector.dart

```diff
@@ -31,39 +31,53 @@ class TextDetector implements FirebaseVisionDetector {
 
     return blocks;
   }
+
+  /// Closes the text detector and release its model resources.
+  @override
+  Future<void> close() {
+    return FirebaseVision.channel.invokeMethod('TextDetector#close');
+  }
 }
 
 /// Abstract class representing dimensions of recognized text in an image.
 abstract class TextContainer {
   TextContainer._(Map<dynamic, dynamic> data)
-      : boundingBox = Rectangle<num>(
-          data['left'],
-          data['top'],
-          data['width'],
-          data['height'],
-        ),
-        cornerPoints = data['points']
+      : boundingBox = data['left'] != null
+            ? Rectangle<num>(
+                data['left'],
+                data['top'],
+                data['width'],
+                data['height'],
+              )
+            : null,
+        _cornerPoints = data['points']
             .map<Point<num>>((dynamic item) => Point<num>(
                   item[0],
                   item[1],
                 ))
             .toList(),
         text = data['text'];
 
-  /// Axis-aligned bounding rectangle of the detected text.
-  final Rectangle<num> boundingBox;
+  final List<Point<num>> _cornerPoints;
```

Comment:
Could we avoid `num` here and just use `double`?

Reply:
tl;dr
num => int

I decided to use `num` because iOS returns `double`s while android returns `int`s. With further investigating though, iOS only returns `double`s for bounding box because `CGRect` uses `double`s. So, I changed num to int to make it clearer to the user that the values will be whole numbers.